### DISCUSSION TITLE
Profile osm reroute.brf

### DISCUSSION THREAD

**Discussion by 0709wiwiwi** [devemux86/cruiser#271] [Q&A] [answered]
Cruiser Brouter reroute.brf profile.

I suggest to add the reroute.brf profile in Cruiser_Brouter.  It allows routing over any nearest available osm path. 
Use so track sections that is not routable by the active profile.  Used as a routing lifesaver when you get stuck. 
Therefore follows any osm path.  It is slightly different from that ultimate drastic rescue mode the beeline.
Mark this track section into the target trkpt src "osm"tag.  Just as is laready implemented with the beeline.

The osm reroute profile is also particularly useful around ferry terminals. 
_These areas often contain numerous obstacles such as access roads tagged as private or destination-only, gates, piers, or inconsistently mapped access restrictions.  As long as the road network **remains connected,** the tolerant nature of the reroute profile often succeeds in finding a practical path where more stricter routing profiles do fail._

**Cruiser Desktop**. 
The desktop version for planning a route is simple and attractive in any case.
But the Cruiser desktop version is little flexible. Beeline is offered when selecting the online BRouter and ORS.
**Update**.  _Desktop v 5.9.8 is the last version that still offers as well the BRouter and ORS online._ ☹️
A similar option that allows to follow any existing **osm** road not hindered by rules would be excellent.
It anyway could definitely be a simple and comfortable **functional** upgrade for the Cruiser desktop version.
Desktop > trackline > right-click > Section profile: | Route profile | Straight line | **osm** (reroute) | 
<img width="278" height="128" alt="Section profile_reroute" src="https://github.com/user-attachments/assets/91ddf830-d239-487f-ab3e-e861ac61e355" />

Add a usefull notation like trkpt src "osm" just as easy as using the trkpt src "beeline"
By this notation as well an osm or beeline tracksections will not be damaged by a "recalculate" all trigger

Reroute tracksection display.
From |SP| to |SP| src reroute.
**------|O|======O|------**

Addtional advantage.
_Prevents OSM data from being too quickly modified without location verification to allow routing in the GPS app being used.
This is a practice that, unfortunately, is sometimes employed, further contaminating OSM data that is already quite unreliable.
If offered in the gps app, this to be used just in VERY EXCEPTIONAL cases, can help to prevent such tempting false OSM edits._
  
Demo file. **Mark “exceptional” indeed as “exceptional”** in the trkpt src osm notation.

[Bike_osm-reroute_Bike.gpx.txt](https://github.com/user-attachments/files/30854855/Bike_osm-reroute_Bike.gpx.txt)

Compare the similarity with the trkpt src beeline notation.  (At a recalculate all action the beeline sections indeed do remain intact)

[Bike_beeline_Bike.gpx.txt](https://github.com/user-attachments/files/30856154/Bike_beeline_Bike.gpx.txt)

The advantage of the highly tolerant osm reroute profile is that it follows any existing OSM path while still generating usefull turn-by-turn (TBT) instructions.  Without OSM path is available, the beeline ultimately remains the only option. When possible, following OSM paths with TBT guidance by the osm reroute option is preferred. 

<img width="1438" height="861" alt="Exeptional" src="https://github.com/user-attachments/assets/77d93578-72fc-42b5-bd8b-41818c95257b" />

Given that possibly only once access allowance on April 1st, you can exceptionally make excellent use of it.

Info RouteYou web.
[Route You. How to optimally use shortest osm and manual (beeline).txt](https://github.com/user-attachments/files/31387477/Route.You.How.to.optimally.use.shortest.osm.and.manual.beeline.txt)

**Comment by devemux86** [comment]
What is the reroute.brf profile?

**Reply by 0709wiwiwi** [reply]
[reroute.brf.txt](https://github.com/devemux86/cruiser/files/12253910/reroute.brf.txt)

Plenty of choice here:
https://bikerouter.de/#map=18/51.06243/13.76567/standard&profile=reroute-zossebart
https://www.mtb-news.de/forum/t/bikerouter-de-brouter-web-fragen-antworten-hilfe-profile-tipps-etc.917910/post-18777450

**Reply by devemux86** [reply]
@0709wiwiwi thanks, you probably mean this:
https://github.com/zossebart/brouter-misc/blob/master/reroute.brf

Profiles are different for each travel mode: car, motorcycle, bike, mountain bike, walking, hiking, horse, rail, river, etc.
Each profile uses the OSM tags that are _specific_ for its travel mode.

We need more info about how it works and where it helps?

**Reply by devemux86** [reply]
Do you mean motor routing to use ways that are not allowed, like steps for pedestrians?

**Reply by 0709wiwiwi** [reply]
Similar to the [Route You shortest osm](https://support.routeyou.com/hc/en-us/articles/16222581159452-Overview-of-RouteYou-routing-options#shortest-osm-0-0) profile that is used for the same purpose.

**Reply by devemux86** [reply, marked as answer]
There is no longer a reroute profile on BRouter:
https://github.com/abrensch/brouter/tree/master/misc/profiles2

**Comment by 0709wiwiwi** [comment]
[shortest-osm or retoute](https://help.routeyou.com/en/topic/view/25/routing-options#shortest-osm)
- Standard profile(s).
- Reroute profile.  To be used exceptionally if your design gets stuck using the standard profile it still follows known osm paths.
- Beeline profile.   As the most ultimate solution the beeline (straight line) solution still keeps available.

**Reroute.**
You follow exactly a(ny) path already drawn in osm. 
The router still so generates turn instructions.

**Beeline.**
Through a router bypass you follow a completely free design. 
Through the router bypass the result contains **no turn instructions**.

**Comment by 0709wiwiwi** [comment]
"Reroute" fits very well between the "Normal" and "Straight line" mode.
Place the"reroute"waypoints **exactly** on an mtb track or goat path to follow.
And add in gpx trkpt src **reroute** similar as with gpx trkpt scrc **beeline**.

**Reply by 0709wiwiwi** [reply]
The reroute profile usage fits very well in cooperation with https://github.com/devemux86/cruiser/discussions/477

**Comment by Leeb64** [comment]
Simply create a profile with reroute as an external profile-
Select profile - section calculation!

**Reply by 0709wiwiwi** [reply]
> Simply create a profile with reroute as an external profile.

Was tested before and was the trigger for the idea launch ;-)

Beeline is written away in the target trkpt src. Reroute can therefore also be saved in trkpt src.
Track sections with data in the target trkpt src are to be protected during any recalculate all.
Be able to recalculate a track with a different general route profile but protect the "specials".

Practical planning as follows.
Select the target Shaping Point > Section Profile. 
Choice: " Route profile # Reroute  # Straight line".

![image](https://github.com/devemux86/cruiser/assets/54167201/96018a1f-aa52-48d2-9f3b-2368ac4c021c)

**Reply by WolfgangZ42** [reply]
> @Leeb64: Simply create a profile with reroute as an external profile - select profile - section calculation!

We are planning to take the ACT-PT in the fall. Mainly on gravel pathes. Therefore I like to test the reroute approach, defined a new profile with external BRouter (up to now I use the internal version only), changed to this profile and tried to assign it to a section (1 to destination). I got this error message:
<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/3aea9802-76ab-4c30-afa4-a7d6873845ab"/>

Is it necessary to provide extra .brf files for an external BRouter profile? Where can I read a little bit more about dealing with external BRouter? Thanks for a hint!

**Reply by devemux86** [reply]
@WolfgangZ42 the message says that there are missing routing data, so have to download them?
Or please post the route and the profile you used.

Regarding external profiles, they can be loaded in BRouter advanced settings.
Note: external profiles do not support Cruiser's section routing options.

**Reply by WolfgangZ42** [reply]
I know what this message says - but the datafile is there, it's my "home area" and I use it day by day. 

Now I learnt that each external profile need its own datafiles - right? I wonder why this is necessary ... 
With the "duplicate" .rd5 file it works. Now I can play with it.

**Reply by devemux86** [reply]
External BRouter app needs its own data files (profiles are not related to the data files).

Also if using external BRouter app, make sure to update as it contains many improvements.
Although better to stay with Cruiser's BRouter implementation, calculations are much faster.

**Reply by WolfgangZ42** [reply]
> Although better to stay with Cruiser's BRouter implementation

I fully agree - not only because of the faster calculation. It is simply easier to use. 
I could now do my test of reroute.brf - here 3 screenshots:

<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/e7925227-80da-4f5e-a889-f403653b8fe3"/>  <img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/b80e8edd-8db2-4eb3-92f3-7f85bec02d68"/>  <img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/863cde77-5371-4480-8765-f7e7fc797e57"/>
Route profile _ _ _ _ _ _ _ _ _ _ _ _ _  Straight line _ _ _ _ _ _ _ _ _ _ _ _ _ Reroute profile

So for me reroute is a good alternative to straight line, especially in gravel touring. I therefore vote for @0709wiwiwi's suggestion to incorporate the function into Cruiser's internal BRouter implementation. It's not high priority, because I now know how to handle it. 

And I see the problem of describing it well enough for "normal" users. As "Straight line light", for example? Perhaps as a sub-function of straight line that has to be selected separately?
As already mentioned in this thread, someone may be standing in front of a staircase on their bike at some point (to be mastered with some bikes). 🥴

**Comment by 0709wiwiwi** [comment]
I sadly had to add the next note in the first contribution.

Update. Desktop v 5.9.8 is the last version that offers as well the BRouter and ORS online. ☹️

_Is there a need or demand for something like this? Well I sure think there is.  
And definitely more useful if able to protect the “special” track section to some extent as with a beeline._
Ref.  https://www.mtb-news.de/forum/t/bikerouter-de-brouter-web-fragen-antworten-hilfe-profile-tipps-etc.917910/post-20471305
